build/thirdparty/darwin/Dockerfile

```diff
@@ -13,45 +13,57 @@
 #  limitations under the License.
 
 # Build or fetch the following binaries for darwin and then host them in a tar.gz file in an alpine image
-# - apiserver (build)
+# - kube-apiserver (build)
 # - kubectl (fetch)
 # - etcd (fetch)
 
-FROM golang:1.15 as darwin
-# Install tools
-ENV CGO 0
-ENV DEST /usr/local/kubebuilder/bin/
+FROM golang:1.15 as builder
 
-RUN apt update && \
-    apt install rsync -y && \
-    apt-get install unzip && \
-    go get github.com/jteeuwen/go-bindata/go-bindata && \
-    ( mkdir -p $DEST || echo "" )
+# Version and platform args.
+ARG KUBERNETES_VERSION
+ARG ETCD_VERSION=v3.4.9
+ARG OS=darwin
+ARG ARCH
 
-RUN git clone https://github.com/kubernetes/kubernetes $GOPATH/src/k8s.io/kubernetes --depth=1 -b v1.20.2
+# Tools path.
+ENV DEST=/usr/local/kubebuilder/bin/
 
-WORKDIR /go/src/k8s.io/kubernetes
+# Install dependencies.
+RUN apt update && \
+    apt install unzip rsync -y && \
+    mkdir -p $DEST
 
+# kube-apiserver
+WORKDIR /kubernetes
+RUN git clone https://github.com/kubernetes/kubernetes . --depth=1 -b ${KUBERNETES_VERSION}
+ENV CGO_ENABLED=0
 # Build for linux first otherwise it won't work for darwin - :(
-ENV KUBE_BUILD_PLATFORMS linux/amd64
+ENV KUBE_BUILD_PLATFORMS=linux/amd64
```

Comment:
Does this still hold true? Do we need to build for linux first? Could we add a reference so that out future selfs or other members understand why this to be done? Also, does it need to be for that hardcoded arch?

Reply:
It doesn't, removed.